vuln-reach/src/javascript/lang/symbol_table.rs

```diff
@@ -299,24 +308,39 @@ impl<'a> SymbolTable<'a> {
 
         let name = self.tree.repr_of(node);
 
-        // If the identifier is in a formal parameters list, the declaration
-        // scope is in the sibling "body" field block.
+        // For function parameters, the scope is the function body.
         let mut parent = cursor.goto_parent().unwrap();
         if parent.kind() == "formal_parameters" {
-            // If formal_parameters nodes has a statement_block next sibling, it is a
-            // regular function, otherwise it is an arrow function and the
-            // identifier doesn't belong to a scope.
-
+            // Get function body.
             let grandparent = cursor.clone().goto_parent().unwrap();
             let body = grandparent.child_by_field_name("body").unwrap();
+
             if body.kind() == "statement_block" {
-                let scope_index = *self.scope_indices.get(&body).unwrap();
-                let scope = &self.scopes[scope_index];
+                // Get scope and ensure the identifier is defined in it.
+                let scope = self.get_scope(body).unwrap();
                 assert!(scope.names.iter().any(|&node| self.tree.repr_of(node) == name));
+
                 return Some((scope, node));
+            } else {
+                // Lambda parameters without body do not belong to any scope.
+                //
+                // Example: `(param) => console.log(param);`
+                return None;
```

Comment:
I added an explicit return here because my understanding is that this would happen anyway in this scenario. This might even fix a bug where the variables would be looked up in the parent scope instead?

Let me know if I misunderstood the purpose of the `statement_block` conditional here.

Reply:
Yes, that makes sense. If the identifier looked up is in the middle of `formal_parameters`, that identifier belongs to the function scope (if applicable) and nowhere else. In current `main` we keep going upwards after the conditional but that probably leads to an incorrect lookup.